src/open_inwoner/utils/logging.py

```diff
@@ -0,0 +1,133 @@
+import time
+from collections import OrderedDict
+from datetime import timedelta
+from logging import Handler  # noqa: TID251 -- needed for base class of handler
+from typing import Mapping, Optional, cast
+from urllib.parse import urlparse
+
+from django.utils import timezone
+
+import structlog
+from log_outgoing_requests.compat import format_exception
+from log_outgoing_requests.typing import (
+    AnyLogRecord,
+    ErrorRequestLogRecord,
+    RequestLogRecord,
+    is_request_log_record,
+)
+from requests import PreparedRequest, RequestException, Response
+
+logger = structlog.stdlib.get_logger(__name__)
+
+
+class StructlogOutgoingRequestsHandler(Handler):
+    """Emit an outgoing request log as structured data."""
+
+    def emit(self, record: AnyLogRecord):
+        # skip requests not coming from the library requests
+        if not record or not is_request_log_record(record):
+            return
+
+        from log_outgoing_requests.models import OutgoingRequestsLogConfig
+        from log_outgoing_requests.utils import process_body
+
+        config = cast(OutgoingRequestsLogConfig, OutgoingRequestsLogConfig.get_solo())
+        record = cast(RequestLogRecord, record)
+
+        # check if we're dealing with success or error state
+        exception = record.exc_info[1] if record.exc_info else None
+        if (response := getattr(record, "res", None)) is not None:
+            # we have a response - this is the 'happy' flow (connectivity is okay)
+            request = record.req
+        elif isinstance(exception, RequestException):
+            record = cast(ErrorRequestLogRecord, record)
+            # we have an requests-specific exception
+            request: Optional[PreparedRequest] = exception.request
+            response: Optional[Response] = exception.response  # likely None
```

Comment:
Types for `request` and `response` are declared in this branch but not the `if` branch. After the `if/elif`, which types do they have? It's hard to tell from the annotation. I would declare them before the branching:
```
request: Optional[PreparedRequest] = None
response: Optional[Response] = None

exception = ...
if (response ...
```